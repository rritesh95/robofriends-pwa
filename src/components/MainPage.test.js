import { shallow } from 'enzyme';
import React from 'react';
import MainPage from './MainPage';

describe("render MainPage with isPending false", () => {

    let wrapper;

    beforeEach( () => {
        const mockProps = {
            onRequestRobots : jest.fn(),
            searchField : '',
            robots : [],
            isPending : false
        }

        wrapper = shallow(<MainPage {...mockProps} />);
    })

    it('render MainPage without crashing', () => {
        expect(wrapper).toMatchSnapshot();
    })

    it('filters robots correctly', () => {
        expect(wrapper.instance().filteredRobots()).toEqual([]);

        const mockProps2 = {
            onRequestRobots : jest.fn(),
            searchField : 'john',
            robots : [{
                id : 3,
                name : "John",
                email : "john@gmail.com"
            }],
            isPending : false
        }

        const wrapper2 = shallow(<MainPage {...mockProps2} />);

        expect(wrapper2.instance().filteredRobots()).toEqual([{
            id : 3,
            name : "John",
            email : "john@gmail.com"
        }]);
    })
})

it('render MainPage with isPending true', () => {
    const PendingTrueMock = {
        onRequestRobots : jest.fn(),
        searchField : '',
        robots : [],
        isPending : true
    };

    const wrapperTrue = shallow(<MainPage {...PendingTrueMock} />);
    expect(wrapperTrue).toMatchSnapshot();
})