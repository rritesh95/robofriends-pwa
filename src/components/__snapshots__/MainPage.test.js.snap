// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MainPage with isPending false render MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={[Function]}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;

exports[`render MainPage with isPending true 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <h1>
      Loading
    </h1>
  </Scroll>
</div>
`;

exports[`render MainPage without crashing 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={[Function]}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
